src/webgl/p5.Framebuffer.js

```diff
@@ -0,0 +1,552 @@
+      console.warn(
+        'This environment does not support FLOAT textures. ' +
+          'Falling back to UNSIGNED_BYTE.'
+      );
+      this.format = constants.UNSIGNED_BYTE;
+    }
+    if (!support.float && this.depthFormat === constants.FLOAT) {
+      console.warn(
+        'This environment does not support FLOAT depth textures. ' +
+          'Falling back to UNSIGNED_BYTE.'
+      );
+      this.depthFormat = constants.UNSIGNED_BYTE;
+    }
+    if (!support.halfFloat && this.format === constants.HALF_FLOAT) {
+      console.warn(
+        'This environment does not support HALF_FLOAT textures. ' +
+          'Falling back to UNSIGNED_BYTE.'
+      );
+      this.format = constants.UNSIGNED_BYTE;
+    }
+
+    if (
+      this.channels === constants.RGB &&
+      [constants.FLOAT, constants.HALF_FLOAT].includes(this.format)
+    ) {
+      console.warn(
+        'FLOAT and HALF_FLOAT formats do not work cross-platform with only ' +
+          'RGB channels. Falling back to RGBA.'
+      );
+      this.channels = constants.RGBA;
+    }
+  }
+
+  _recreateTextures() {
+    const gl = this.gl;
+
+    this._updateSize();
+
+    const prevBoundTexture = gl.getParameter(gl.TEXTURE_BINDING_2D);
+    const prevBoundFramebuffer = gl.getParameter(gl.FRAMEBUFFER_BINDING);
+
+    const colorTexture = gl.createTexture();
+    if (!colorTexture) {
+      throw new Error('Unable to create color texture');
+    }
+    gl.bindTexture(gl.TEXTURE_2D, colorTexture);
+    const colorFormat = this._glColorFormat();
+    gl.texImage2D(
+      gl.TEXTURE_2D,
+      0,
+      colorFormat.internalFormat,
+      this.width * this.density,
+      this.height * this.density,
+      0,
+      colorFormat.format,
+      colorFormat.type,
+      null
+    );
+
+    // Create the depth texture
+    const depthTexture = gl.createTexture();
+    if (!depthTexture) {
+      throw new Error('Unable to create depth texture');
+    }
+    const depthFormat = this._glDepthFormat();
+    gl.bindTexture(gl.TEXTURE_2D, depthTexture);
+    gl.texImage2D(
+      gl.TEXTURE_2D,
+      0,
+      depthFormat.internalFormat,
+      this.width * this.density,
+      this.height * this.density,
+      0,
+      depthFormat.format,
+      depthFormat.type,
+      null
+    );
+
+    gl.bindFramebuffer(gl.FRAMEBUFFER, this.framebuffer);
+    gl.framebufferTexture2D(
+      gl.FRAMEBUFFER,
+      gl.COLOR_ATTACHMENT0,
+      gl.TEXTURE_2D,
+      colorTexture,
+      0
+    );
+    gl.framebufferTexture2D(
+      gl.FRAMEBUFFER,
+      gl.DEPTH_ATTACHMENT,
+      gl.TEXTURE_2D,
+      depthTexture,
+      0
+    );
+
+    // Create separate framebuffer for antialiasing
+    if (this.antialias) {
+      this.colorRenderbuffer = gl.createRenderbuffer();
+      gl.bindRenderbuffer(gl.RENDERBUFFER, this.colorRenderbuffer);
+      gl.renderbufferStorageMultisample(
+        gl.RENDERBUFFER,
+        Math.min(4, gl.getParameter(gl.MAX_SAMPLES)),
```

Comment:
Do we want to expose the ability to set the number of samples?

Reply:
We could potentially just keep the `antialias` settings option the same, but default to 2 or 4 if its true, or if someone includes a number it would use that instead.
``` 
settings = { antialias: false }; // no antialiasing
settings = { antialias: true }; // 4x msaa
settings = { antialias: 8 }; // 8x msaa
settings = { antialias: 0 }; // no antialiasing
settings = { antialias: 100000 }; // Math.min(gl.getParameter(gl.MAX_SAMPLES), settings.antialias)
``` 

Also, do you think 4 is overkill, or does 2 look ok? Might be best to start with a lowest value that produces decent results as a default.

Reply:
I like that API design! Also 2 seems to look decent to me, so especially if there's a way to change it, defaulting to that makes sense.